### PR TITLE
DBZ-5917 Fix querying column metadata for tables with back slash in t…

### PR DESCRIPTION
**Pull request by vjuranek**
…he name

If the table contains back slash, which is ANSI SQL escape chracter, in its name, querying column metadata would fail in some cases (*) which would result into NPE. Fix table name before the query and escape escape character if the table name contains it.

Default backslash works for all currently supported databases as it's ANSI SQL standard, but in case of need it escape  methods can be overridded respective subclasses.

(*) everything works when we collect column metadata for all tables, i.e. when there are no excluded tables

https://issues.redhat.com/browse/DBZ-5917

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java [hunk 1]
```diff
@@ -1223,11 +1224,23 @@ protected String resolveCatalogName(String catalogName) {
         return catalogName;
     }
 
+    protected static CharSequence getEscapeSequence() {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java:1227]: @vjuranek Is not this a bit overkill? Cant' we just use `String` constant `DEFAULT_ESCAPE_CHAR`?

Comment (vjuranek) [reply debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java:1227]: It's here for the purpose we can override it in DB specific sub-classes if there is any other escape char. But given that `\` is SQL standard, it's questionable if we will ever needed to use it. So yes, let's use simple approach and add possibility to override it when we really need it.

**Review by jpechane — APPROVED** [review]
@vjuranek Could you please resolve the conflict? Feel free to merge the PR yourself after that.

**Comment by vjuranek** [comment]
Test failures are not related.

**Pull request merged**